### PR TITLE
make clinical attributes search/filter case insensitive

### PR DESCRIPTION
**Pull request by khzhu**
# What? Why?
Fix issue [https://github.com/cBioPortal/cbioportal/issues/5853](url)

Changes proposed in this pull request:
- a
- b

# Checks
- [ ] Is this PR adding logic based on one or more **clinical** attributes? If yes, please make sure validation for this attribute is also present in the data validation / data loading layers (in backend repo) and documented in [File-Formats Clinical data section](https://github.com/cBioPortal/cbioportal/blob/master/docs/File-Formats.md#clinical-data)!
- [ ] Follows [7 rules of great commit messages](http://chris.beams.io/posts/git-commit/). For most PRs a single commit should suffice, in some cases multiple topical commits can be useful. During review it is ok to see tiny commits (e.g. Fix reviewer comments), but right before the code gets merged to master or rc branch, any such commits should be squashed since they are useless to the other developers. Definitely avoid [merge commits, use rebase instead.](http://nathanleclaire.com/blog/2014/09/14/dont-be-scared-of-git-rebase/)
- [ ] Follows the [Airbnb React/JSX Style guide](https://github.com/airbnb/javascript/tree/master/react).
- [ ] Make sure your commit messages end with a Signed-off-by string (this line
  can be automatically added by git if you run the `git-commit` command with
  the `-s` option)

# Any screenshots or GIFs?
If this is a new visual feature please add a before/after screenshot or gif
here with e.g. [GifGrabber](http://www.gifgrabber.com/).

# Notify reviewers
Read our [Pull request merging
policy](../CONTRIBUTING.md#pull-request-merging-policy). If you are part of the
cBioPortal organization, notify the approprate team (remove inappropriate):

@cBioPortal/frontend

If you are not part of the cBioPortal organization look at who worked on the
file before you. Please use `git blame <filename>` to determine that
and notify them here:

### CODE REVIEW

**Comment by khzhu** [comment]
Hi @zhx828 @kalletlak , please finish your review when you have time. Thanks!

**Comment by khzhu** [comment]
@zhx828 , that was what originally thought. But, I got an error for not converting to upper case before a comparison (please see attached).
Screen Shot 2019-09-18 at 5 08 33 PM

**Comment by zhx828** [comment]
> @zhx828 , that was what originally thought. But, I got an error for not converting to upper case before a comparison (please see attached).
> Screen Shot 2019-09-18 at 5 08 33 PM

@khzhu this seems related to the clinical attribute id not the value. The first upper case you did for clinical attribute id should solve the issue. But the second upper case is unnecessary.

**Comment by khzhu** [comment]
@zhx828 , you were right. removed the second change, still working. thanks!

**Comment by khzhu** [comment]
@zhx828 , could you please take a final look at the code? Thanks!

File: src/pages/studyView/StudyViewPageStore.ts [hunk 1]
```diff
@@ -822,7 +822,7 @@ export class StudyViewPageStore {
                 studyIds: studyIds
             }), clinicalAttribute => `${clinicalAttribute.patientAttribute}-${clinicalAttribute.clinicalAttributeId}`);
 
-            const matchedAttr = _.find(clinicalAttributes, (attr: ClinicalAttribute) => attr.clinicalAttributeId === query.filterAttributeId);
+            const matchedAttr = _.find(clinicalAttributes, (attr: ClinicalAttribute) => attr.clinicalAttributeId.toUpperCase() === query.filterAttributeId!.toLocaleString().toUpperCase());
```
[the remark this answers is not in the document]
Comment (khzhu) [reply src/pages/studyView/StudyViewPageStore.ts]: agree. removed and testing now.

[the remark this answers is not in the document]
Comment (khzhu) [reply src/pages/studyView/StudyViewPageStore.ts]: worked. can I please have your approval, @zhx828, thanks!

**Pull request merged**